docs/v3tov4.md

```diff
@@ -0,0 +1,283 @@
+# Upgrading from GoSDK v3.x to GoSDK v4.x
+
+Support for older versions have been removed. Therefore, this new version is not compatible with the previous one. 
+This document describes the changes between versions and how to adapt an integration built with the GoSDK v3.x to the 
+new one.
+
+The Go SDK v4 contains the following changes:
+
+* A new JSON schema version 4.
+* Add support for dimensional metrics using the [Metrics API format][1].
+* New metric data types: `count` and `summary`.
+* Removed support for `RATE`, `PRATE` and `DELTA` metric types.
+* Local entities have been removed.
+* Metrics, Events and Inventory can be attached either to Entities or an Integration.
+* Support for Go Modules
+* Removed support for protocols prior to v4.x.
+
+## JSON schema changes
+
+The following section explains the new JSON schema. Be aware that the new SDK only supports this new JSON document. 
+These are the most important changes:
+
+* A new "integration" object at the top-level.
+
+* The "entity" and "metrics" objects have been modified.
+
+### JSON protocol v4 sample
+```
+{
+  "protocol_version":"4",                      # protocol version number
+  "integration":{                              # this data will be added to all metrics and events as attributes,                                               
+                                               # and also sent as inventory.
+    "name":"integration name",
+    "version":"integration version"
+  },
+  "data":[                                    # List of entities
+    {
+      "metrics":[                             # list of metrics using the dimensional metric format
+        {
+          "name":"redis.metric1",
+          "type":"count",                     # gauge, count, summary or pdelta
+          "value":93, 
+          "attributes":{}                     # set of key-value pairs that define the dimensions of the metric
```

Comment:
why do we call this attributes and not dimensions?

Reply:
we follow the same naming convention defined in the Metric API https://docs.newrelic.com/docs/data-ingest-apis/get-data-new-relic/metric-api/report-metrics-metric-api#json-payload-keys